### PR TITLE
Add a toggle for hiding mangoAccount in leaderboards + search

### PR DESCRIPTION
**Pull request by Lou-Kamades**

### CODE REVIEW

**Comment by saml33** [comment]
Cool. I started this today as well (without the api). Think the setting should go under "Account" so it only shows when a user has an account.

Not sure how you think this should work but I assumed we'd have another endpoint to retrieve all of the private accounts and use the list to block from viewing via url and leaderboard. I've done this with mock data here: https://github.com/blockworks-foundation/mango-v4-ui/tree/private-accounts

**Comment by Lou-Kamades** [comment]
I agree with you regarding the extra endpoint and how we should block accounts from showing up. I just deployed th endpoint for all private accounts at: https://api.mngo.cloud/data/v4/user-data/private-accounts. Take a look and lmk if this or the other endpoints or request signing doesn't make sense.

I can leave the design/ui work to you for this feature if you'd prefer to do it, I got a little eager with this PR bc I wanted to POC and play with react-query which I'd never used before hahaha

File: components/account/HideMangoAccount.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  return (
+    <>
+      <div className="flex items-center justify-between border-t border-th-bkg-3 p-4">
+        <p className="">Hide Mango Account From Search</p>
+        {signingForHide ? (
+          <Loading />
+        ) : (
+          <Switch
+            checked={accountHidden ?? false}
+            onChange={async () => {
```
Comment (saml33) [inline components/account/HideMangoAccount.tsx:26]: This function should probably be defined as a constant and have a try/catch block

File: components/account/AccountHeroStats.tsx [hunk 2]
```diff
@@ -345,17 +353,35 @@ const AccountHeroStats = ({
...
-            </p>
-            <div className="flex space-x-1.5">
-              <Change change={rollingDailyPnlChange} prefix="$" size="small" />
-              <p className="text-xs text-th-fgd-4">{t('rolling-change')}</p>
-            </div>
+            {!accountHidden ? (
+              <>
+                <p className="mb-0.5 mt-1 text-left text-2xl font-bold text-th-fgd-1 lg:text-xl xl:text-2xl">
+                  <FormatNumericValue
+                    value={accountPnl}
```
Comment (saml33) [inline components/account/AccountHeroStats.tsx:360]: Can use accountHidden in the component here so we don't repeat the jsx i.e. value={!accountHidden ? accountPnl : 0}

File: components/account/AccountHeroStats.tsx [hunk 3]
```diff
@@ -345,17 +353,35 @@ const AccountHeroStats = ({
...
+                    decimals={2}
+                    isUsd={true}
+                  />
+                </p>
+                <div className="flex space-x-1.5">
+                  <Change
+                    change={rollingDailyPnlChange}
```
Comment (saml33) [inline components/account/AccountHeroStats.tsx:367]: Can use accountHidden in the component here

File: hooks/useMangoAccountHidden.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      requestOptions,
+    )
+  } catch (e) {
+    if (isMangoError(e)) {
+      notify({
+        title: 'idk',
+        description: e.message,
+        txid: e.txid,
+        type: 'error',
+      })
```
Comment (tlrsssss) [inline hooks/useMangoAccountHidden.ts:82]: This doesn't look right. I think `isMangoError` only works for v4 client functions. Does this api return a txid for the error?

**Review by tlrsssss — CHANGES_REQUESTED** [review]

**Review by tlrsssss — APPROVED** [review]

**Pull request merged**